src/lab11.py

```diff
@@ -481,38 +498,92 @@ def raster_tab(self):
         canvas.clear(skia.ColorWHITE)
         active_tab.raster(canvas)
 
-    # We need to keep this because it calls get_font
+    def plus_bounds(self):
+        plus_width = self.chrome_font.measureText("+")
+        return (self.padding, self.padding,
+            plus_width + self.padding, self.tab_header_bottom - self.padding)
```

Comment:
I guess my main aesthetic objection is that we're "polluting" the `Browser` class with a bunch more methods so it's harder to glance at the code and say "what does `Browser`, fundamentally, do?" I'm not sure if it'd be better to move all of this to a `Chrome` class, but that seems like the other option.

Reply:
Potentially the `Chrome` class could have a `layout` and `paint` method? (`layout` would compute the plus / tab / backbutton / urlbar bounds and store them in fields, `paint` would read those fields?) That might be a nice way to talk about the similarity between chrome and web content without actually doing HTML-y layout.